### PR TITLE
Fixed #18119 -- Added a DomainNameValidator validator.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/18119

### CODE REVIEW

File: django/core/validators.py [hunk 1]
```diff
@@ -238,6 +238,20 @@ def __eq__(self, other):
 
 validate_email = EmailValidator()
 
+
+@deconstructible
+class DomainNameValidator(RegexValidator):
+    message = _('Enter a valid domain name value.')
+    regex = _lazy_re_compile(
+        r'(?:' +
+        URLValidator.ipv4_re + '|' +
```
Comment (timgraham) [inline django/core/validators.py]: Are the ip re's included by mistake? They aren't tested.

**Comment by timgraham** [comment]
Did you consider Claude's comment about [using the validator elsewhere](https://code.djangoproject.com/ticket/18119#comment:1)?

**Comment by pypetey** [comment]
@berkerpeksag @timgraham  - this issue seems to be inactive. If there's a need I could complete the PR and fix remaining conflicts.

File: django/core/validators.py (hunk 1 above)
Comment (orzel) [reply django/core/validators.py]: IPs can be used in URL, so I guess that's what they mean here. Doesn't shock me.

**Comment by Lamberk** [comment]
Hi! Am i right that we still can not validate domain names?

**Comment by sanjioh** [comment]
I'd love to see this landed as well :)

File: django/core/validators.py (hunk 1 above)
Comment (sanjioh) [reply django/core/validators.py]: Hi, I was looking for a domain validator in Django and I've found this PR.

I think a domain name should also be checked against a max length, as `URLValidator` does (https://github.com/django/django/blob/master/django/core/validators.py#L137).

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**